src/anomalib/pipelines/components/actions/grid_search.py

```diff
@@ -0,0 +1,162 @@
+"""Utils for benchmarking."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+from collections.abc import Generator
+from itertools import product
+from typing import Any
+
+from jsonargparse import Namespace
+from jsonargparse._typehints import ActionTypeHint
+
+from anomalib.pipelines.components import (
+    convert_to_tuple,
+    dict_from_namespace,
+    flatten_dict,
+    namespace_from_dict,
+    to_nested_dict,
```

Comment:
are these pipeline specific implementations? could they go somewhere in `anomalib.utils`. I remember there are some similar util functions somewhere else in the repo. nncf maybe